earthfile2llb/converter.go

```diff
@@ -2306,6 +2307,10 @@ func (c *Converter) internalRun(ctx context.Context, opts ConvertRunOpts) (pllb.
 	// Shell and debugger wrap.
 	prependDebugger := !opts.Locally
 	finalArgs = opts.shellWrap(finalArgs, extraEnvVars, opts.WithShell, prependDebugger, isInteractive)
+	if opts.NoCache {
+		// llb.IgnoreCache is not always enough; we will force a different cache key as a work-around
+		finalArgs = append(finalArgs, "#"+uuid.NewString())
```

Comment:
Should this be used in all the `WITH DOCKER` cases too? Or in `LOCALLY` too?

Any other situations where we do a run underneath but it's not part of the RUN command? (IF? FOR? - don't think those have --no-cache variants though)

Reply:
good question -- it does work `WITH DOCKER`, which calls `internalRun`.

e.g. https://github.com/earthly/earthly/blob/7ad30b8d90034448b2796e1328f9fa875dd30375/earthfile2llb/with_docker_run_local_reg.go#L120

it's run in all cases:

```
$ grep internalRun earthfile2llb/with_docker_*.go
earthfile2llb/with_docker_run_local_reg.go:	_, err = w.c.internalRun(ctx, crOpts)
earthfile2llb/with_docker_run_local_tar.go:	_, err = w.c.internalRun(ctx, crOpts)
earthfile2llb/with_docker_run_reg.go:	_, err = w.c.internalRun(ctx, crOpts)
earthfile2llb/with_docker_run_tar.go:	_, err = w.c.internalRun(ctx, crOpts)
```

the `--no-cache` has no effect on `LOCALLY`, but adding it doesn't hurt.

We actually need this for `IF`, `FOR`, which do support the `--no-cache` option.

Reply:
oh interesting! yeah maybe it'll solve the COPY+LOCALLY issues :crossed_fingers: